### DISCUSSION TITLE
Kubernetes 1.26 support

### DISCUSSION THREAD

**Discussion by mhamzarazzaq97** [fabric8io/kubernetes-client#5132] [Q&A]
When and In which release version Kubernetes 1.26 will be supported?

**Comment by rohanKanojia** [comment]
Do you mean 1.26 or 1.27? Support for 1.26 was added in https://github.com/fabric8io/kubernetes-client/pull/4731 . It should be available in versions >= 6.5.0

**Comment by manusa** [comment]
Kubernetes 1.26 types are already supported.

We'll add Kuberentes **1.27** specific types in next release, you can track this in #5056.

In any case, you should be able to interact with **any** Kubernetes Cluster (newer or legacy) with any of the latest releases of the client. For unsupported types you'll get `GenericKubernetesResource` instead of a dedicated type.

**Reply by mhamzarazzaq97** [reply]
@manusa @rohanKanojia , Thanks for your answer,
But in the readme, the compatibility matrix is saying that Kubernetes 1.26 is only supported on HEAD. Is readme file is outdated? If yes, please update the readme file, it will clear confusion for others like me.
<img width="938" alt="image" src="https://github.com/fabric8io/kubernetes-client/assets/52797825/4863dd8a-93b0-418a-b50f-0a64f0e58278">

**Reply by manusa** [reply]
This compatibility matrix just shows the types included. I think we definitely need to remove it since it is very misleading and produces more confusion than help.

Thoughts @rohanKanojia @sunix @shawkins @metacosm

**Reply by rohanKanojia** [reply]
Umm, I think I forgot to update the compatibility matrix after the release of client. 

If we remove this, what alternative should we choose to show kubernetes API compatibility to users?

**Reply by manusa** [reply]
Something in the lines of:

---

## Compatibility

The Fabric8 Kubernetes Client is compatible with **any** standard Kubernetes Cluster. The client also provides Java model types for most of the Kubernetes objects and resources (If you encounter there's some type missing, please file an issue). For any Kubernetes object that's not recognized by the client when performing an operation, instead of returning the specific Java type instance, a `GenericKubernetesResource` or `RawExtension` instance will be returned.

### OpenShift

The Fabric8 OpenShift and Kubernetes Client is compatible with **any** supported OpenShift cluster. The OpenShift client also provides Java model types for most of the OpenShift-specific objects and resources (If you encounter there's some type missing, please file an issue).

---

**Reply by rohanKanojia** [reply]
okay, let's remove this table and add this instead.

**Reply by metacosm** [reply]
What does `supported OpenShift` mean in this context? I'm fine with removing the table as well. I would either add a link to the documentation on `GenericKubernetesResource` / `RawExtension` or add examples here so that people can quickly solve their issues without having to look too long for a solution.

**Reply by manusa** [reply]
> What does supported OpenShift mean in this context?

I meant any OpenShift cluster version actively supported by Red Hat. So maybe change to "The Fabric8 OpenShift and Kubernetes Client is compatible with any OpenShift cluster version currently supported by Red Hat.

> I would either add a link to the documentation...

I'm not sure we have those properly documented. Could only find these:
- https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md#resource-api
- https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md#resource-typeless-api

So maybe something to improve too.

**Reply by mhamzarazzaq97** [reply]
> We'll add Kuberentes **1.27** specific types in next release

Thanks for the clarification @manusa . 
Need one more clarification, as you said here that kubernetes 1.27 specific type will be added in next release, does this mean that we can use fabric8 version 6.6.2 with kubernetes version 1.27 but the new types that are introduced in 1.27 will not be there in version 6.6.2 ? 
or does this mean, we cannot use fabric8 version 6.6.2 with kubernetes version 1.27 at all ?
Thanks

**Reply by manusa** [reply]
> that we can use fabric8 version 6.6.2 with kubernetes version 1.27 but the new types that are introduced in 1.27 will not be there in version 6.6.2

^ this

You can safely use the Fabric8 Client with it (or any other version)

You won't have builders or types for any of the very specific new types that are introduced in 1.27. For the most common scenarios it would be strange that you'd need any of those. Hiowever, if necessary, you can work around those by using `GenericKubernetesResource`.
